Source/Http/ServiceRequest.swift

```diff
@@ -58,146 +58,145 @@ class ServiceRequest {
-                if response.response != nil {
+            case .failure(var error):
+				if response.response != nil {
                     if let data = response.data {
-                        error = Error.requestErrorWithData(data)
+						error = SparkError.requestErrorWith(data: data)
                     }
                 }
-                result = .Failure(error)
+                result = .failure(error)
             }
             
             completionHandler(ServiceResponse(response.response, result))
         }
     }
     
-    func responseJSON(completionHandler: ServiceResponse<AnyObject> -> Void){
-        let requset = Alamofire.request(URLRequest()).validate()
+    func responseJSON(_ completionHandler: @escaping (ServiceResponse<Any>) -> Void){
+        let request = createAlamofireRequest()
         
-        requset.responseJSON(queue: queue){
-            (response: Response<AnyObject, NSError>) in
-            var result: Result<AnyObject>
+        request.responseJSON(queue: queue){
+            (response: DataResponse<Any>) in
+            var result: Result<Any>
             
             switch response.result {
-            case .Success(let value):
-                result = .Success(value)
+            case .success(let value):
+                result = .success(value)
                 
-            case .Failure(var error):
+            case .failure(var error):
                 if response.response != nil {
                     if let data = response.data {
-                        error = Error.requestErrorWithData(data)
+						error = SparkError.requestErrorWith(data: data)
                     }
                 }
-                result = .Failure(error)
+                result = .failure(error)
             }
             
             completionHandler(ServiceResponse(response.response, result))
         }
     }
     
-    private func URLRequest() -> NSMutableURLRequest {
-        var request = NSMutableURLRequest(URL: url)
-        request.HTTPMethod = method.rawValue
-        
-        if authRequired {
-            if let authorization = AuthManager.sharedInstance.getAuthorization() {
-                headers.unionInPlace(authorization)
-            }
-        }
-        
-        for (field, value) in headers {
-            request.setValue(value, forHTTPHeaderField: field)
-        }
-        
-        if let body = body {
-            request = ParameterEncoding.JSON.encode(request, parameters: body.value()).0
-        }
-        if let query = query {
-            request = ParameterEncoding.URL.encode(request, parameters: query.value()).0
-        }
-        
-        return request
-    }
+	private func createAlamofireRequest() -> Alamofire.DataRequest {
+		if authRequired {
+			if let authorization = AuthManager.sharedInstance.getAuthorization() {
+				headers.unionInPlace(authorization)
+			}
+		}
+		let parameters: [String:Any]?
+		let encoding: ParameterEncoding
+		if let body = body {
```

Comment:
The original code support JSON and URL encoding both in one request. The code change will break it.